### ISSUE TITLE
File::Spec->catpath ignoring tlg_dir on macOS

### ISSUE THREAD

**Issue by ben-larson** [pjheslin/diogenes#80]
When I search for an author in TLG Texts corpus, I get an error page.

<img width="572" alt="image" src="https://user-images.githubusercontent.com/10101828/139363146-1a7e979a-5814-4ae1-ac1e-852df7a2ca86.png">

## Steps to reproduce:

- OS: macOS 11.6
- Running server via `server/diogenes-server.pl -d -P ~/Documents/TLG-E`
- Try to search for an author in TLG Texts corpus ([example link](http://localhost:8889/Diogenes.cgi?JumpTo=&JumpFromQuery=&JumpFromLang=&JumpFromAction=&action=author&splash=true&export-path=&corpus=TLG+Texts&author=aristoteles&query=&go=Go&current_pageXXstate=splash&greek_output_formatXXstate=UTF-8))

## What I think the problem is

_Disclaimer: not a perl developer_

There was a commit called **Fix chron search filename paths on Windows.** (https://github.com/pjheslin/diogenes/commit/e1126a4c200f03fe0c4dcfae276a89700b4e2f7b) which removed `$vol` from a call to `File::Spec->catpath`.  I think this broke unix systems, because the [documentation](https://perldoc.perl.org/File::Spec#catpath()) for `File::Spec` says:

> Under Unix, `$volume` is ignored, and directory and file are concatenated.

I think it expects the volume to be the first of three arguments, so it just ignores the first one on unix systems.  Based on my logging, only the file name is being returned by `catpath`.

```perl
File::Spec->catpath($self->{tlg_dir}, $filename); # tlg9007.txt
File::Spec->catpath($vol, $self->{tlg_dir}, $filename); # /Users/benlarson/Documents/TLG-E/tlg7052.txt
```

When I put `$vol` back, it works.

Please let me know if there's more context I can provide, or if I'm just confused and this is just due to misconfiguration on my end.

**Comment by ben-larson** [comment]
Using an empty string as the volume works on macOS, so I put in pull request #81 for that.  I do not have access to Linux or Windows systems to test it out, so I left it as a draft pull request, but if it works on those then it is ready for review.

**Comment by pjheslin** [comment]
Thanks for the fantastic bug report and fix!  I'm sorry it has taken me so long to get back to you -- it has been a very busy university term.   

I was fixing a problem with Windows users where the volume appeared in the path twice.  Removing  the volume from the call to `catpath` fixed it on Windows, but broke it on Mac.  Your fix is correct and I have merged it.  I should have supplied an empty volume instead.

The reason I did not catch this before is that it typically does not break searching the TLG; it just breaks chronological searching, which is a very new feature.  I'm not sure why it broke things more badly for you.

Note that the -P argument for the server is obsolete and was never used to indicate the location of the TLG.  It was used once upon a time for telling the server where the Perseus data was located.
